smqtk_iqr/web/iqr_service/iqr_server.py

```diff
@@ -188,6 +196,15 @@ def __init__(self, json_config: Dict):
 
         # Initialize from config
         self.positive_seed_neighbors = sc_config['positive_seed_neighbors']
+        self.autoneg_select_ratio = sc_config['autoneg_select_ratio']
+        if sc_config['distance_metric'] == 'euclidean':
+            self.distance_metric: Union[Callable[[np.ndarray, np.ndarray], np.ndarray],
+                                        Callable[[np.ndarray, np.ndarray], float]] = euclidean_distance
+        elif sc_config['distance_metric'] == 'cosine':
+            self.distance_metric = cosine_distance
+        elif sc_config['distance_metric'] == 'histogram_intersection':
+            self.distance_metric = histogram_intersection_distance
```

Comment:
We can avoid an large if-else statement and incorporate invalid-value error emission by creating a mapping instead of the if-switch. The map could be defined on the global level for visibility, but just in here would be fine for now.
```suggestion
        metric_map: Dict[str, Callable[[np.ndarray, np.ndarray], np.ndarray]] = {
            "euclidean": euclidean_distance,
            "cosine": cosine_distance,
            "histogram_intersection": histogram_intersection_distance,
        }
        self.distance_metric = metric_map[sc_config['distance_metric']]
```
The `KeyError` may even be caught to translate it into a more informative error for the user.

Reply:
Would it be OK to catch the error and then assign the `distance_metric` to be the `euclidean_distance` function or should the server error out in some way?

Reply:
I missed this reply. Generally, we should stay away from implicit defaulting unless we have a specific reason to be doing so. I don't know why we would want to do that here, so I generally suggest falling back to the "error hard and fast" philosophy.